app/src/main/java/com/example/concordia_campus_guide/Activities/PathsActivity.java

```diff
@@ -130,19 +132,18 @@ public void checkFromToType(Place from, Place to) {
 
     public void setIndoorPaths() {
         if(!fromIsIndoor && !toIsIndoor) return;
-        String floorCode;
-        Place tempFrom = from;
-        Place tempTo = to;
 
-        if(fromIsIndoor && !toIsIndoor){
-            floorCode = ((RoomModel) from).getFloorCode();
-            tempTo = new RoomModel(from.getCenterCoordinates(), floorCode.substring(0, floorCode.indexOf('-')), floorCode, from.getCampus());
+        //Outdoor -> Indoor
+        if (!fromIsIndoor) {
+            locationFragment.setIndoorPaths(mViewModel.getEntrance(to), to);
         }
-        if(toIsIndoor && !fromIsIndoor){
-            floorCode = ((RoomModel) to).getFloorCode();
-            tempFrom = new RoomModel(to.getCenterCoordinates(), floorCode.substring(0, floorCode.indexOf('-')), floorCode, to.getCampus());
+        else if(mViewModel.arePlacesSeparatedByATunnel(from, to) || mViewModel.areInSameBuilding(from, to)){
+            locationFragment.setIndoorPaths(from, to);
+        }
+        //[from -> from_entrance ] + outdoor directions + [to_entrance -> to]
```

Comment:
Do we need to keep this ?

Reply:
Yes, so that the purpose of the code is clear (at least it helped me).